eventloop/queue.go

```diff
@@ -21,13 +25,13 @@ func newQueue(capacity uint) queue {
 	}
 }
 
-func (q *queue) push(entry any) {
+// push adds an entry to the buffer in a FIFO fashion. If the queue is full, the first
+// entry is dropped to make space for the newest entry and returns true.
+func (q *queue) push(entry any) (droppedOne bool) {
```

Comment:
What if we instead return the dropped event; then we can log the event that was dropped. See implementation below.
```go
func (q *queue) push(entry any) (droppedEvent any) {
...
		droppedEvent = q.entries[q.head] // Line 46 with current: dropped = true
...
	return droppedEvent
```